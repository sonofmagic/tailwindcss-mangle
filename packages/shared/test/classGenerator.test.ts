import ClassGenerator from '@/classGenerator'

describe('classGenerator', () => {
  it('size 26*27+1', () => {
    const result: string[] = []
    const clsGen = new ClassGenerator()
    for (let i = 0; i < 26 * 27 + 1; i++) {
      result.push(clsGen.defaultClassGenerate())
      clsGen.newClassSize++
    }
    expect(result).toMatchSnapshot()
  })

  it('26*27+1', () => {
    const clsGen = new ClassGenerator()
    clsGen.newClassSize = 26 * 27
    expect(clsGen.defaultClassGenerate()).toBe('tw-aaa')
  })
})
