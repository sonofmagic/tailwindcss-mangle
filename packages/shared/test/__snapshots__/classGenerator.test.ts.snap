// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`classGenerator > size 26*27+1 1`] = `
[
  "tw-a",
  "tw-b",
  "tw-c",
  "tw-d",
  "tw-e",
  "tw-f",
  "tw-g",
  "tw-h",
  "tw-i",
  "tw-j",
  "tw-k",
  "tw-l",
  "tw-m",
  "tw-n",
  "tw-o",
  "tw-p",
  "tw-q",
  "tw-r",
  "tw-s",
  "tw-t",
  "tw-u",
  "tw-v",
  "tw-w",
  "tw-x",
  "tw-y",
  "tw-z",
  "tw-aa",
  "tw-ba",
  "tw-ca",
  "tw-da",
  "tw-ea",
  "tw-fa",
  "tw-ga",
  "tw-ha",
  "tw-ia",
  "tw-ja",
  "tw-ka",
  "tw-la",
  "tw-ma",
  "tw-na",
  "tw-oa",
  "tw-pa",
  "tw-qa",
  "tw-ra",
  "tw-sa",
  "tw-ta",
  "tw-ua",
  "tw-va",
  "tw-wa",
  "tw-xa",
  "tw-ya",
  "tw-za",
  "tw-ab",
  "tw-bb",
  "tw-cb",
  "tw-db",
  "tw-eb",
  "tw-fb",
  "tw-gb",
  "tw-hb",
  "tw-ib",
  "tw-jb",
  "tw-kb",
  "tw-lb",
  "tw-mb",
  "tw-nb",
  "tw-ob",
  "tw-pb",
  "tw-qb",
  "tw-rb",
  "tw-sb",
  "tw-tb",
  "tw-ub",
  "tw-vb",
  "tw-wb",
  "tw-xb",
  "tw-yb",
  "tw-zb",
  "tw-ac",
  "tw-bc",
  "tw-cc",
  "tw-dc",
  "tw-ec",
  "tw-fc",
  "tw-gc",
  "tw-hc",
  "tw-ic",
  "tw-jc",
  "tw-kc",
  "tw-lc",
  "tw-mc",
  "tw-nc",
  "tw-oc",
  "tw-pc",
  "tw-qc",
  "tw-rc",
  "tw-sc",
  "tw-tc",
  "tw-uc",
  "tw-vc",
  "tw-wc",
  "tw-xc",
  "tw-yc",
  "tw-zc",
  "tw-ad",
  "tw-bd",
  "tw-cd",
  "tw-dd",
  "tw-ed",
  "tw-fd",
  "tw-gd",
  "tw-hd",
  "tw-id",
  "tw-jd",
  "tw-kd",
  "tw-ld",
  "tw-md",
  "tw-nd",
  "tw-od",
  "tw-pd",
  "tw-qd",
  "tw-rd",
  "tw-sd",
  "tw-td",
  "tw-ud",
  "tw-vd",
  "tw-wd",
  "tw-xd",
  "tw-yd",
  "tw-zd",
  "tw-ae",
  "tw-be",
  "tw-ce",
  "tw-de",
  "tw-ee",
  "tw-fe",
  "tw-ge",
  "tw-he",
  "tw-ie",
  "tw-je",
  "tw-ke",
  "tw-le",
  "tw-me",
  "tw-ne",
  "tw-oe",
  "tw-pe",
  "tw-qe",
  "tw-re",
  "tw-se",
  "tw-te",
  "tw-ue",
  "tw-ve",
  "tw-we",
  "tw-xe",
  "tw-ye",
  "tw-ze",
  "tw-af",
  "tw-bf",
  "tw-cf",
  "tw-df",
  "tw-ef",
  "tw-ff",
  "tw-gf",
  "tw-hf",
  "tw-if",
  "tw-jf",
  "tw-kf",
  "tw-lf",
  "tw-mf",
  "tw-nf",
  "tw-of",
  "tw-pf",
  "tw-qf",
  "tw-rf",
  "tw-sf",
  "tw-tf",
  "tw-uf",
  "tw-vf",
  "tw-wf",
  "tw-xf",
  "tw-yf",
  "tw-zf",
  "tw-ag",
  "tw-bg",
  "tw-cg",
  "tw-dg",
  "tw-eg",
  "tw-fg",
  "tw-gg",
  "tw-hg",
  "tw-ig",
  "tw-jg",
  "tw-kg",
  "tw-lg",
  "tw-mg",
  "tw-ng",
  "tw-og",
  "tw-pg",
  "tw-qg",
  "tw-rg",
  "tw-sg",
  "tw-tg",
  "tw-ug",
  "tw-vg",
  "tw-wg",
  "tw-xg",
  "tw-yg",
  "tw-zg",
  "tw-ah",
  "tw-bh",
  "tw-ch",
  "tw-dh",
  "tw-eh",
  "tw-fh",
  "tw-gh",
  "tw-hh",
  "tw-ih",
  "tw-jh",
  "tw-kh",
  "tw-lh",
  "tw-mh",
  "tw-nh",
  "tw-oh",
  "tw-ph",
  "tw-qh",
  "tw-rh",
  "tw-sh",
  "tw-th",
  "tw-uh",
  "tw-vh",
  "tw-wh",
  "tw-xh",
  "tw-yh",
  "tw-zh",
  "tw-ai",
  "tw-bi",
  "tw-ci",
  "tw-di",
  "tw-ei",
  "tw-fi",
  "tw-gi",
  "tw-hi",
  "tw-ii",
  "tw-ji",
  "tw-ki",
  "tw-li",
  "tw-mi",
  "tw-ni",
  "tw-oi",
  "tw-pi",
  "tw-qi",
  "tw-ri",
  "tw-si",
  "tw-ti",
  "tw-ui",
  "tw-vi",
  "tw-wi",
  "tw-xi",
  "tw-yi",
  "tw-zi",
  "tw-aj",
  "tw-bj",
  "tw-cj",
  "tw-dj",
  "tw-ej",
  "tw-fj",
  "tw-gj",
  "tw-hj",
  "tw-ij",
  "tw-jj",
  "tw-kj",
  "tw-lj",
  "tw-mj",
  "tw-nj",
  "tw-oj",
  "tw-pj",
  "tw-qj",
  "tw-rj",
  "tw-sj",
  "tw-tj",
  "tw-uj",
  "tw-vj",
  "tw-wj",
  "tw-xj",
  "tw-yj",
  "tw-zj",
  "tw-ak",
  "tw-bk",
  "tw-ck",
  "tw-dk",
  "tw-ek",
  "tw-fk",
  "tw-gk",
  "tw-hk",
  "tw-ik",
  "tw-jk",
  "tw-kk",
  "tw-lk",
  "tw-mk",
  "tw-nk",
  "tw-ok",
  "tw-pk",
  "tw-qk",
  "tw-rk",
  "tw-sk",
  "tw-tk",
  "tw-uk",
  "tw-vk",
  "tw-wk",
  "tw-xk",
  "tw-yk",
  "tw-zk",
  "tw-al",
  "tw-bl",
  "tw-cl",
  "tw-dl",
  "tw-el",
  "tw-fl",
  "tw-gl",
  "tw-hl",
  "tw-il",
  "tw-jl",
  "tw-kl",
  "tw-ll",
  "tw-ml",
  "tw-nl",
  "tw-ol",
  "tw-pl",
  "tw-ql",
  "tw-rl",
  "tw-sl",
  "tw-tl",
  "tw-ul",
  "tw-vl",
  "tw-wl",
  "tw-xl",
  "tw-yl",
  "tw-zl",
  "tw-am",
  "tw-bm",
  "tw-cm",
  "tw-dm",
  "tw-em",
  "tw-fm",
  "tw-gm",
  "tw-hm",
  "tw-im",
  "tw-jm",
  "tw-km",
  "tw-lm",
  "tw-mm",
  "tw-nm",
  "tw-om",
  "tw-pm",
  "tw-qm",
  "tw-rm",
  "tw-sm",
  "tw-tm",
  "tw-um",
  "tw-vm",
  "tw-wm",
  "tw-xm",
  "tw-ym",
  "tw-zm",
  "tw-an",
  "tw-bn",
  "tw-cn",
  "tw-dn",
  "tw-en",
  "tw-fn",
  "tw-gn",
  "tw-hn",
  "tw-in",
  "tw-jn",
  "tw-kn",
  "tw-ln",
  "tw-mn",
  "tw-nn",
  "tw-on",
  "tw-pn",
  "tw-qn",
  "tw-rn",
  "tw-sn",
  "tw-tn",
  "tw-un",
  "tw-vn",
  "tw-wn",
  "tw-xn",
  "tw-yn",
  "tw-zn",
  "tw-ao",
  "tw-bo",
  "tw-co",
  "tw-do",
  "tw-eo",
  "tw-fo",
  "tw-go",
  "tw-ho",
  "tw-io",
  "tw-jo",
  "tw-ko",
  "tw-lo",
  "tw-mo",
  "tw-no",
  "tw-oo",
  "tw-po",
  "tw-qo",
  "tw-ro",
  "tw-so",
  "tw-to",
  "tw-uo",
  "tw-vo",
  "tw-wo",
  "tw-xo",
  "tw-yo",
  "tw-zo",
  "tw-ap",
  "tw-bp",
  "tw-cp",
  "tw-dp",
  "tw-ep",
  "tw-fp",
  "tw-gp",
  "tw-hp",
  "tw-ip",
  "tw-jp",
  "tw-kp",
  "tw-lp",
  "tw-mp",
  "tw-np",
  "tw-op",
  "tw-pp",
  "tw-qp",
  "tw-rp",
  "tw-sp",
  "tw-tp",
  "tw-up",
  "tw-vp",
  "tw-wp",
  "tw-xp",
  "tw-yp",
  "tw-zp",
  "tw-aq",
  "tw-bq",
  "tw-cq",
  "tw-dq",
  "tw-eq",
  "tw-fq",
  "tw-gq",
  "tw-hq",
  "tw-iq",
  "tw-jq",
  "tw-kq",
  "tw-lq",
  "tw-mq",
  "tw-nq",
  "tw-oq",
  "tw-pq",
  "tw-qq",
  "tw-rq",
  "tw-sq",
  "tw-tq",
  "tw-uq",
  "tw-vq",
  "tw-wq",
  "tw-xq",
  "tw-yq",
  "tw-zq",
  "tw-ar",
  "tw-br",
  "tw-cr",
  "tw-dr",
  "tw-er",
  "tw-fr",
  "tw-gr",
  "tw-hr",
  "tw-ir",
  "tw-jr",
  "tw-kr",
  "tw-lr",
  "tw-mr",
  "tw-nr",
  "tw-or",
  "tw-pr",
  "tw-qr",
  "tw-rr",
  "tw-sr",
  "tw-tr",
  "tw-ur",
  "tw-vr",
  "tw-wr",
  "tw-xr",
  "tw-yr",
  "tw-zr",
  "tw-as",
  "tw-bs",
  "tw-cs",
  "tw-ds",
  "tw-es",
  "tw-fs",
  "tw-gs",
  "tw-hs",
  "tw-is",
  "tw-js",
  "tw-ks",
  "tw-ls",
  "tw-ms",
  "tw-ns",
  "tw-os",
  "tw-ps",
  "tw-qs",
  "tw-rs",
  "tw-ss",
  "tw-ts",
  "tw-us",
  "tw-vs",
  "tw-ws",
  "tw-xs",
  "tw-ys",
  "tw-zs",
  "tw-at",
  "tw-bt",
  "tw-ct",
  "tw-dt",
  "tw-et",
  "tw-ft",
  "tw-gt",
  "tw-ht",
  "tw-it",
  "tw-jt",
  "tw-kt",
  "tw-lt",
  "tw-mt",
  "tw-nt",
  "tw-ot",
  "tw-pt",
  "tw-qt",
  "tw-rt",
  "tw-st",
  "tw-tt",
  "tw-ut",
  "tw-vt",
  "tw-wt",
  "tw-xt",
  "tw-yt",
  "tw-zt",
  "tw-au",
  "tw-bu",
  "tw-cu",
  "tw-du",
  "tw-eu",
  "tw-fu",
  "tw-gu",
  "tw-hu",
  "tw-iu",
  "tw-ju",
  "tw-ku",
  "tw-lu",
  "tw-mu",
  "tw-nu",
  "tw-ou",
  "tw-pu",
  "tw-qu",
  "tw-ru",
  "tw-su",
  "tw-tu",
  "tw-uu",
  "tw-vu",
  "tw-wu",
  "tw-xu",
  "tw-yu",
  "tw-zu",
  "tw-av",
  "tw-bv",
  "tw-cv",
  "tw-dv",
  "tw-ev",
  "tw-fv",
  "tw-gv",
  "tw-hv",
  "tw-iv",
  "tw-jv",
  "tw-kv",
  "tw-lv",
  "tw-mv",
  "tw-nv",
  "tw-ov",
  "tw-pv",
  "tw-qv",
  "tw-rv",
  "tw-sv",
  "tw-tv",
  "tw-uv",
  "tw-vv",
  "tw-wv",
  "tw-xv",
  "tw-yv",
  "tw-zv",
  "tw-aw",
  "tw-bw",
  "tw-cw",
  "tw-dw",
  "tw-ew",
  "tw-fw",
  "tw-gw",
  "tw-hw",
  "tw-iw",
  "tw-jw",
  "tw-kw",
  "tw-lw",
  "tw-mw",
  "tw-nw",
  "tw-ow",
  "tw-pw",
  "tw-qw",
  "tw-rw",
  "tw-sw",
  "tw-tw",
  "tw-uw",
  "tw-vw",
  "tw-ww",
  "tw-xw",
  "tw-yw",
  "tw-zw",
  "tw-ax",
  "tw-bx",
  "tw-cx",
  "tw-dx",
  "tw-ex",
  "tw-fx",
  "tw-gx",
  "tw-hx",
  "tw-ix",
  "tw-jx",
  "tw-kx",
  "tw-lx",
  "tw-mx",
  "tw-nx",
  "tw-ox",
  "tw-px",
  "tw-qx",
  "tw-rx",
  "tw-sx",
  "tw-tx",
  "tw-ux",
  "tw-vx",
  "tw-wx",
  "tw-xx",
  "tw-yx",
  "tw-zx",
  "tw-ay",
  "tw-by",
  "tw-cy",
  "tw-dy",
  "tw-ey",
  "tw-fy",
  "tw-gy",
  "tw-hy",
  "tw-iy",
  "tw-jy",
  "tw-ky",
  "tw-ly",
  "tw-my",
  "tw-ny",
  "tw-oy",
  "tw-py",
  "tw-qy",
  "tw-ry",
  "tw-sy",
  "tw-ty",
  "tw-uy",
  "tw-vy",
  "tw-wy",
  "tw-xy",
  "tw-yy",
  "tw-zy",
  "tw-az",
  "tw-bz",
  "tw-cz",
  "tw-dz",
  "tw-ez",
  "tw-fz",
  "tw-gz",
  "tw-hz",
  "tw-iz",
  "tw-jz",
  "tw-kz",
  "tw-lz",
  "tw-mz",
  "tw-nz",
  "tw-oz",
  "tw-pz",
  "tw-qz",
  "tw-rz",
  "tw-sz",
  "tw-tz",
  "tw-uz",
  "tw-vz",
  "tw-wz",
  "tw-xz",
  "tw-yz",
  "tw-zz",
  "tw-aaa",
]
`;
